source/grid/tria.cc

```diff
@@ -2196,41 +2196,14 @@ namespace internal
                                                       "manifold ids"));
             line->set_manifold_id(subcell_line.manifold_id);
 
-            // Assert that only exterior lines are given a boundary
-            // indicator; however, it is possible that someone may
-            // want to give an interior line a manifold id (and thus
-            // lists this line in the subcell_data structure), and we
-            // need to allow that
-            if (subcell_line.boundary_id != numbers::internal_face_boundary_id)
-              {
-                if (line->boundary_id() == numbers::internal_face_boundary_id)
-                  {
-                    // if we are here, it means that we want to assign a
-                    // boundary indicator different from
-                    // numbers::internal_face_boundary_id to an internal line.
-                    // As said, this would be not allowed, and an exception
-                    // should be immediately thrown. Still, there is the
-                    // possibility that one only wants to specify a manifold_id
-                    // here. If that is the case (manifold_id !=
-                    // numbers::flat_manifold_id) the operation is allowed.
-                    // Otherwise, we really tried to specify a boundary_id (and
-                    // not a manifold_id) to an internal face. The exception
-                    // must be thrown.
-                    if (subcell_line.manifold_id == numbers::flat_manifold_id)
-                      {
-                        // If we are here, this assertion will surely fail, for
-                        // the aforementioned reasons
-                        AssertThrow(!(line->boundary_id() ==
-                                      numbers::internal_face_boundary_id),
-                                    ExcInteriorLineCantBeBoundary(
-                                      line->vertex_index(0),
-                                      line->vertex_index(1),
-                                      subcell_line.boundary_id));
-                      }
-                  }
-                else
-                  line->set_boundary_id_internal(subcell_line.boundary_id);
-              }
+            // Assert that only exterior lines are given a boundary indicator
+            AssertThrow(
+              subcell_line.boundary_id == numbers::internal_face_boundary_id ||
+                line->boundary_id() != numbers::internal_face_boundary_id,
```

Comment:
Thanks for cleaning this up.

I might be missing something, but I am not convinced that this assertion is correct. I think it is possible for it to pass when `subcell_line.boundary_id == numbers::internal_face_boundary_id` and `line->boundary_id() != numbers::internal_face_boundary_id`. If so then we just assigned a boundary line to have an interior id. Is this right?

Reply:
Yes, you're right. It should read
```
if (subcell_line.boundary_id != numbers::internal_face_boundary_id)
  {
    AssertThrow(
      line->boundary_id() != numbers::internal_face_boundary_id,
      ExcInteriorLineCantBeBoundary(line->vertex_index(0),
                                    line->vertex_index(1),
                                    subcell_line.boundary_id));
    line->set_boundary_id_internal(subcell_line.boundary_id);
  }
```
I fixed that now.